astronomer/providers/amazon/aws/triggers/emr.py

```diff
@@ -71,6 +71,125 @@ async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Serializes EmrContainerOperatorTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.amazon.aws.triggers.emr.EmrContainerOperatorTrigger",
+            {
+                "virtual_cluster_id": self.virtual_cluster_id,
+                "name": self.name,
+                "execution_role_arn": self.execution_role_arn,
+                "release_label": self.release_label,
+                "job_driver": self.job_driver,
+                "configuration_overrides": self.configuration_overrides,
+                "client_request_token": self.client_request_token,
+                "aws_conn_id": self.aws_conn_id,
+                "max_tries": self.max_tries,
+                "poll_interval": self.poll_interval,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """Run until EMR container reaches the desire state"""
+        hook = EmrContainerHookAsync(aws_conn_id=self.aws_conn_id, virtual_cluster_id=self.virtual_cluster_id)
+        try:
+            job_id = await hook.submit_job(
+                name=self.name,
+                execution_role_arn=self.execution_role_arn,
+                release_label=self.release_label,
+                job_driver=self.job_driver,
+                configuration_overrides=self.configuration_overrides,
+                client_request_token=self.client_request_token,
+            )
+
+            poll: int = 1
+            while True:
+                query_status = await hook.check_job_status(job_id)
+                if query_status in self.INTERMEDIATE_STATES or query_status is None:
+                    if self.max_tries and poll >= self.max_tries:
+                        yield TriggerEvent(
+                            {
+                                "status": "error",
+                                "message": "Timeout: Maximum retry limit exceed",
+                                "job_id": job_id,
+                            }
+                        )
+                        return
+                    poll += 1
+                    self.log.info(
+                        "Job status is %s. Sleeping for %s seconds.", query_status, self.poll_interval
+                    )
+                    await asyncio.sleep(self.poll_interval)
+                elif query_status in self.FAILURE_STATES:
+                    yield TriggerEvent(
+                        {
+                            "status": "error",
+                            "message": f"EMR Containers Operator failed {query_status}",
+                            "job_id": job_id,
+                        }
+                    )
+                    return
+                elif query_status in self.SUCCESS_STATES:
+                    yield TriggerEvent(
+                        {
+                            "status": "success",
+                            "message": f"EMR Containers Operator success {query_status}",
```

Comment:
what are the logs being shown on the sync version of the operator? We need to send same logs from here to the worker